src/adapter/z-stack/znp/znp.ts

```diff
@@ -138,28 +139,28 @@ class Znp extends events.EventEmitter {
 
         this.unpiParser = new UnpiParser();
         this.serialPort.pipe(this.unpiParser);
-        this.unpiParser.on('parsed', this.onUnpiParsed);
+        this.unpiParser.on('parsed', this.onUnpiParsed.bind(this));
 
-        return new Promise((resolve, reject): void => {
-            this.serialPort.open(async (error): Promise<void> => {
-                if (error) {
-                    reject(new Error(`Error while opening serialport '${error}'`));
-                    this.initialized = false;
-                    if (this.serialPort.isOpen) {
-                        this.serialPort.close();
-                    }
-                } else {
-                    debug.log('Serialport opened');
-                    this.initialized = true;
-                    await this.skipBootloader();
-                    this.serialPort.once('close', this.onPortClose);
-                    this.serialPort.once('error', (error) => {
-                        debug.error(`Serialport error: ${error}`);
-                    });
-                    resolve();
-                }
-            });
-        });
+        try {
+            await this.serialPort.asyncOpen();
+            debug.log('Serialport opened');
+
+            this.serialPort.once('close', this.onPortClose.bind(this));
+            this.serialPort.once('error', this.onPortError.bind(this));
+
+            this.initialized = true;
+            this.emit('connected');
```

Comment:
I noticed this has been added but the event doesn't seem to be used anywhere (same for `ezsp/driver/uart.ts` where it was already present)

Reply:
Yes it could be removed entirely indeed, if there is no intent to use it; probably leftover code, but I went conservative on removals at first.

Reply:
Let's remove it